scalacheck/src/main/scala/pathy/scalacheck/PathOf.scala

```diff
@@ -54,19 +54,14 @@ object PathOf {
     genRelFile[A] map (rootDir </> _)
 
   private def genRelFile[A: Arbitrary: Show]: Gen[RelFile[Sandboxed]] =
-    for {
-      d <- genRelDir[A]
-      s <- genSegment[A]
-    } yield d </> file(s)
+    sizeDistributed(genRelDir[A], genSegment[A])((d, s) => d </> file(s))
```

Comment:
Yes. If you just use `nonEmptyListOf` as we did before, you get lists of up to `size` segments, each with up to `size` chars. The problem is that by default scalacheck/specs2 start with small sizes and then go as high as 100/200, which means paths of thousands of chars. On the other hand, you don't want to override the size to something hard-coded because it means you don't get the nice behavior of failing on nice small values first.

The idea of this "aggregate" size stuff is for `size` to control the overall size. For example, with 0, this always generates `/` (for absDir), with 1, you get `/a/`, 2 -> `/ab/`, 3 -> `/abc/` or `/a/b/`, etc. And the paths grow linearly with the size param, which means no "reasonable" size will generate an unreasonably large path.